image-classification-with-distributed-training/horovod-project.ipynb

```diff
@@ -5,92 +5,203 @@
+    "\n",
+    "![Horovod scaling](https://user-images.githubusercontent.com/16640218/38965607-bf5c46ca-4332-11e8-895a-b9c137e86013.png)\n",
+    "\n",
+    "### How can we distribute our training\n",
+    "There are two different cluster configurations (which can be combined) we need to take into account.  \n",
+    "- **Multi Node** &mdash; GPUs are distributed over multiple nodes in the cluster.  \n",
+    "- **Multi GPU** &mdash; GPUs are within a single Node.  \n",
+    "\n",
+    "In this demo we show a **Multi Node Multi GPU** &mdash; **Data Parallel** enabled training using Horovod.  \n",
+    "However, you should always try and use the best distribution strategy for your use case (due to the added costs of the distribution itself, ability to run in an optimized way on specific hardware or other considerations that may arise).\n",
+    "\n",
+    "### How Horovod works?\n",
+    "Horovod's primary motivation is to make it easy to take a single-GPU training script and successfully scale it to train across many GPUs in parallel. This has two aspects:\n",
+    "\n",
+    "- How much modification does one have to make to a program to make it distributed, and how easy is it to run it?\n",
+    "- How much faster would it run in distributed mode?\n",
+    "\n",
+    "Horovod Supports TensorFlow, Keras, PyTorch, and Apache MXNet.\n",
+    "\n",
+    "in MLRun we use Horovod with MPI in order to create cluster resources and allow for optimized networking.  \n",
+    "**Note:** Horovd and MPI may use [NCCL](https://developer.nvidia.com/nccl) when applicable which may require some specific configuration arguments to run optimally.\n",
+    "\n",
+    "Horovod uses this MPI and NCCL concepts for distributed computation and messaging to quickly and easily synchronize between the different nodes or GPUs.\n",
+    "\n",
+    "![Ring Allreduce Strategy](https://miro.medium.com/max/700/1*XdMlfmOgPCUG9ZOYLTeP9w.jpeg)\n",
+    "\n",
+    "Horovod will run your code on all the given nodes (Specific node can be addressed via `hvd.rank()`) while using an `hvd.DistributedOptimizer` wrapper to run the **synchronization cycles** between the copies of your Neural Network running at each node.  \n",
+    "\n",
+    "**Note:** Since all the copies of your Neural Network must be the same, Your workers will adjust themselves to the rate of the slowest worker (simply by waiting for it to finish the epoch and receive its updates). Thus try not to make a specific worker do a lot of additional work on each epoch (Like a lot of saving, extra calculations, etc...) since this can affect the overall training time.\n",
+    "\n",
+    "\n",
+    "### How do we integrate TF2 with Horovod?\n",
+    "As it's one of the main motivations, integration is fairly easy and requires only a few steps: ([You can read the full instructions for all the different frameworks on Horovod's documentation website](https://horovod.readthedocs.io/en/stable/tensorflow.html))\n",
```

Comment:
Maybe also mention we use ResNet50V2 as a baseline, add classification layers and then add Horovod for distributed training

Reply:
Added a paragraph above the MPIJob function definition